nft-minter/src/nft_attributes_builder.rs

```diff
@@ -0,0 +1,168 @@
+        let collection_id = self.parent_collection_id().get();
+        let id_ascii = self.decimal_to_ascii(nft_id as u32);
+
+        let mut metadata = GenericAttributes::new_from_bytes(METADATA_PREFIX);
+        metadata.append(&collection_id);
+        metadata.append_bytes(SLASH);
+        metadata.append(&id_ascii);
+        metadata.append_bytes(DOT);
+        metadata.append_bytes(JSON_FILE_EXTENSION);
+
+        metadata
+    }
+
+    fn build_attributes_tags_part(
+        &self,
+        brand_id: &BrandId<Self::Api>,
+    ) -> GenericAttributes<Self::Api> {
+        let all_tags = self.tags_for_brand(brand_id).get();
+        let tags_len = all_tags.len();
+        if tags_len == 0 {
+            return GenericAttributes::new();
+        }
+
+        let mut tags_attributes = GenericAttributes::new_from_bytes(TAGS_PREFIX);
+        for i in 0..tags_len - 1 {
+            let tag = all_tags.get(i);
+            tags_attributes.append(&tag);
+            tags_attributes.append_bytes(TAG_SEPARATOR);
+        }
+
+        let last_tag = all_tags.get(tags_len - 1);
+        tags_attributes.append(&last_tag);
+
+        tags_attributes
+    }
+
+    fn build_nft_main_file_uri(
+        &self,
+        nft_id: UniqueId,
+        media_type: &MediaType<Self::Api>,
+    ) -> Uri<Self::Api> {
+        let mut uri = self.build_base_uri_for_id(nft_id);
+        uri.append(media_type);
+
+        uri
+    }
+
+    fn build_nft_json_file_uri(&self, nft_id: UniqueId) -> Uri<Self::Api> {
+        let mut uri = self.build_base_uri_for_id(nft_id);
+        uri.append_bytes(JSON_FILE_EXTENSION);
+
+        uri
+    }
+
+    fn build_collection_json_file_uri(&self) -> Uri<Self::Api> {
+        let mut uri = self.build_base_collection_uri();
+        uri.append_bytes(COLLECTION_INFO_FILE_NAME);
+        uri.append_bytes(DOT);
+        uri.append_bytes(JSON_FILE_EXTENSION);
+
+        uri
+    }
+
+    fn build_base_uri_for_id(&self, nft_id: UniqueId) -> Uri<Self::Api> {
+        let id_ascii = self.decimal_to_ascii(nft_id as u32);
+
+        let mut uri = self.build_base_collection_uri();
+        uri.append(&id_ascii);
+        uri.append_bytes(DOT);
+
+        uri
+    }
+
+    fn build_base_collection_uri(&self) -> Uri<Self::Api> {
+        let collection_id = self.parent_collection_id().get();
+
+        let mut uri = Uri::new_from_bytes(BASE_URI);
+        uri.append_bytes(SLASH);
+        uri.append(&collection_id);
+        uri.append_bytes(SLASH);
+
+        uri
+    }
+
+    fn is_supported_media_type(&self, media_type: &MediaType<Self::Api>) -> bool {
+        let media_type_len = media_type.len();
+        if media_type_len > MAX_MEDIA_TYPE_LEN {
+            return false;
+        }
+
+        let mut media_static_buffer = [0u8; MAX_MEDIA_TYPE_LEN];
+        let slice = &mut media_static_buffer[..media_type_len];
+        let _ = media_type.load_slice(0, slice);
+
+        SUPPORTED_MEDIA_TYPES.contains(&&slice[..])
```

Comment:
clippy

Reply:
Clippy is actually wrong this time, using `slice` directly causes an error. I've added an annotation for clippy to ignore this.